src/shadowbox/server/manager_metrics.ts

```diff
@@ -15,7 +15,10 @@
 import {PrometheusClient} from '../infrastructure/prometheus_scraper';
 import {DataUsageByUser} from '../model/metrics';
 
-export interface ManagerMetrics { get30DayByteTransfer(): Promise<DataUsageByUser>; }
+export interface ManagerMetrics {
+  get30DayByteTransfer(): Promise<DataUsageByUser>;
+  getOutboundByteTransfer(accessKeyId: string, windowHours: number): Promise<DataUsageByUser>;
```

Comment:
This seems to violate the single responsibility rule. I don't see a good reason to conflate the metrics we show to the manager with the quota enforcement system.

Let's separate those things.

Maybe you can place the logic in the `ServerAccessKeyRepository`. Or in an abstraction that lives there. You could inject the Prometheus client in the constructor, rather than the ManagerMetrics.

BTW, the main reason ManagerMetrics exists is because of the transition from the old to the new metrics when we migrated to Go.

Reply:
It looked like the right place to me, since it already reports outbound metrics, but I see your point in that it serves manager-facing metrics. Implemented an injectable abstraction `UsageMetrics` in 2a653b4.